### PR TITLE
Remove electron configuration

### PR DESCRIPTION
**Pull request by Aerijo**
### Description of the Change

Removes the environment configuration concerning the electron version. Managing the environment is apm's job, and it [already sets this variable anyway](https://github.com/atom/apm/blob/ff9170e2efaf11949dd9e194b0f6f3e09d37569c/src/command.coffee#L110).

### Alternate Designs

None. Only apm should be managing electron compatibility logic.

### Possible Drawbacks

None. It has no effect now anyway because apm overwrites it.

### Verification Process

CI passes

### Release Notes

NA

### CODE REVIEW

File: script/lib/install-script-dependencies.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const childProcess = require('child_process');
 
 const CONFIG = require('../config');
 
+// For the 'electron-mksnapshot' dependency
```
Comment (DeeDeeG) [inline script/lib/install-script-dependencies.js:7]: > // For the `electron-mksnapshot` dependency

This is also for `electron-chromedriver`.

File: script/lib/run-apm-install.js [hunk 2]
```diff
@@ -8,9 +8,7 @@ module.exports = function(packagePath, ci, stdioOptions) {
   const installEnv = Object.assign({}, process.env);
   // Set resource path so that apm can load metadata related to Atom.
   installEnv.ATOM_RESOURCE_PATH = CONFIG.repositoryRootPath;
-  // Set our target (Electron) version so that node-pre-gyp can download the
-  // proper binaries.
-  installEnv.npm_config_target = CONFIG.appMetadata.electronVersion;
+
```
Comment (DeeDeeG) [inline script/lib/run-apm-install.js:11]: 👍 

apm should indeed be setting this very same env var here:

https://github.com/atom/apm/blob/1384050001726904ee00b30b596df57e0169e131/src/command.coffee#L110

Comment (DeeDeeG) [reply script/lib/run-apm-install.js:11]: Oops, I see you already linked this same line of source code. My bad.

File: script/lib/install-script-dependencies.js (hunk 1 above)
Comment (sadick254) [reply script/lib/install-script-dependencies.js:7]: @Aerijo could you add `electron-chromedriver` to the comment.

**Review by sadick254 — COMMENTED** [review]
Looks great. 👍

File: script/lib/install-script-dependencies.js (hunk 1 above)
Comment (Aerijo) [reply script/lib/install-script-dependencies.js]: I've added it, but I couldn't find any direct reference to Chromedriver using it (then again, I couldn't find their complete changelogs / source either). The closest I've found is https://github.com/electron/chromedriver/pull/52#issuecomment-630845509.

Comment (DeeDeeG) [reply script/lib/install-script-dependencies.js]: It's briefly mentioned in the `README.md`, and I've read the source code and found that it is indeed used.

https://github.com/electron/chromedriver#overriding-the-version-downloaded

`electron-mksnapshot` and `electron-chromedriver` use `@electron/get`.
https://github.com/electron/chromedriver/blob/3d0e75f88efe5e7026ef47e0c6997e25196ea14d/download-chromedriver.js#L3-L8

`@electron/get` respects `ELECTRON_CUSTOM_VERSION`.

- https://github.com/electron/chromedriver/blob/3d0e75f88efe5e7026ef47e0c6997e25196ea14d/download-chromedriver.js#L3-L8
- https://github.com/electron/get/blob/7b295f33d20cfafe556cdb81c56b3e6d006a9ea6/README.md#overriding-the-version-downloaded

This was all tested/confirmed during the electron 6 PR here at the main Atom repo. (Was merged separately from Electron 6). https://github.com/atom/atom/pull/20879

Comment (Aerijo) [reply script/lib/install-script-dependencies.js]: Ah thanks, I thought they were talking about the Chromedriver project itself in that comment. I didn't think to check `@electron/get`. I've updated the comment to reflect what is actually recognising that variable, which should help guard against it becoming out of date.

**Pull request merged**